satpy/tests/reader_tests/test_fci_l1c_nc.py

```diff
@@ -103,18 +103,21 @@ def _get_test_calib_for_channel_ir(data, meas_path):
     from pyspectral.blackbody import C_SPEED as c
     from pyspectral.blackbody import H_PLANCK as h
     from pyspectral.blackbody import K_BOLTZMANN as k
-    data[meas_path + "/radiance_to_bt_conversion_coefficient_wavenumber"] = FakeH5Variable(da.array(955))
-    data[meas_path + "/radiance_to_bt_conversion_coefficient_a"] = FakeH5Variable(da.array(1))
-    data[meas_path + "/radiance_to_bt_conversion_coefficient_b"] = FakeH5Variable(da.array(0.4))
-    data[meas_path + "/radiance_to_bt_conversion_constant_c1"] = FakeH5Variable(da.array(1e11 * 2 * h * c ** 2))
-    data[meas_path + "/radiance_to_bt_conversion_constant_c2"] = FakeH5Variable(da.array(1e2 * h * c / k))
+    data[meas_path + "/radiance_to_bt_conversion_coefficient_wavenumber"] = FakeH5Variable(
+        da.array(955.0, dtype=np.float32))
+    data[meas_path + "/radiance_to_bt_conversion_coefficient_a"] = FakeH5Variable(da.array(1.0, dtype=np.float32))
+    data[meas_path + "/radiance_to_bt_conversion_coefficient_b"] = FakeH5Variable(da.array(0.4, dtype=np.float32))
+    data[meas_path + "/radiance_to_bt_conversion_constant_c1"] = FakeH5Variable(
+        da.array(1e11 * 2 * h * c ** 2, dtype=np.float32))
+    data[meas_path + "/radiance_to_bt_conversion_constant_c2"] = FakeH5Variable(
+        da.array(1e2 * h * c / k, dtype=np.float32))
```

Comment:
Is there a specific reason why you had to fix the type of the coefficients here? Now that the coefficients type casting is hardcoded inside the reader, I would assume that what we want to test now is rather that the final output is `float32`, no matter the type of the inputs that come from the files? In other words, if one would remove the type casting inside the reader, the tests would still pass?
(Commenting here, but it applies to most changes below too)

Reply:
I was initially trying to mimic the dtypes of the real data files, so converted these coefficients to `float32`. Because the `FakeH5Variable` doesn't mimic exactly the` behaviour of the `netCDF4` (or `h5netcdf`) libraries, the coeffs are collected by a different branch of the code. I wrote the description in a haste earlier today, but examples there are exactly for this case.

The dtypes in the `FakeH5Variable` can act as a reminder of the actual data types for the brave developer who starts to convert the FCI tests to use a stub file instead of these classes.